var/spack/repos/builtin/packages/axl/package.py

```diff
@@ -47,12 +33,9 @@ def cmake_args(self):
             args.append("-DAXL_LINK_STATIC=ON")
         args.append("-DWITH_KVTREE_PREFIX=%s" % self.spec['kvtree'].prefix)
 
-        apis = list(self.spec.variants['async_api'].value)
-        if 'daemon' in apis:
-            args.append('-DAXL_ASYNC_DAEMON=ON')
-            apis.remove('daemon')
-
-        for api in apis:
-            args.append('-DAXL_ASYNC_API={0}'.format(api))
+        if '+bbapi_fallback' in self.spec:
+            args.append('-DENABLE_BBAPI_FALLBACK=ON')
```

Comment:
Do the older versions make use of this option too, or would they need to use `-DAXL_ASYNC_API`?

Reply:
No, the older versions didn't have the `-DENABLE_BBAPI_FALLBACK=ON` option.
However, the `-DAXL_ASYNC_API` is also broken in those versions, other than the default. AXL automatically gets set to default in those versions when not passing this option though. So they still wouldn't need the `-DAXL_ASYNC_API` option.

Hence why we decided the best option was to just deprecate all the older versions. However, do I still need to write Spack packages to support deprecated versions?

Reply:
Hmm, yeah, just looked over the Spack docs again on deprecating older version, and it looks like we should still maintain at least more recent versions in the event someone is still trying to build with them.

I'll add a conditional for the `-DENABLE_BBAPI_FALLBACK=ON` option to only append with v0.4.0, and `-DAXL_ASYNC_API` with older versions.

Reply:
You can also remove the older version (similar to the conversation at https://github.com/spack/spack/pull/22390#discussion_r616244097)

Reply:
Pushed an update.

I would have needed to remove all the old versions for this one since we're phasing out of the `async_api` variant.
So instead, I updated the older versions using the `async_api` variant to still build with `-DAXL_ASYNC_API`, but to only be used with those versions.
Then set `-DENABLE_BBAPI_FALLBACK` to only be used with the newest version.